kernel/src/table_features/mod.rs

```diff
@@ -221,6 +701,331 @@ pub(crate) static SUPPORTED_WRITER_FEATURES: LazyLock<Vec<TableFeature>> = LazyL
     ]
 });
 
+/// A unified interface for checking table feature support and enablement.
+///
+/// This struct combines Protocol and TableProperties to provide a single source of truth for:
```

Comment:
Some features require things like checking the table schema (i.e. iceberg compat). I think we're gonna want the metadata in here too?